documentation/docs/30-advanced/70-packaging.md

```diff
@@ -2,33 +2,105 @@
+
+### license
+
+Every package should have a license field so people know how they are allowed to use it. A very popular license which is also very permissive in terms of distribution and reuse without warranty is `MIT`.
+
+```json
+{
+	"license": "MIT"
+}
+```
+
+Read more about it [here](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#license). Note that you should also include a `LICENSE` file in your package.
+
+### files
+
+This tells npm which files it will pack up and upload to npm. It should contain your output folder (`dist` by default). Your `package.json` and `README` and `LICENSE` will always be included, so you don't need to specify them.
+
+```json
+{
+	"files": ["dist"]
+}
+```
 
-- All the files in `src/lib`, unless you [configure](configuration) custom `include`/`exclude` options. Svelte components will be preprocessed, TypeScript files will be transpiled to JavaScript.
-- Type definitions (`d.ts` files) which are generated for Svelte, JavaScript and TypeScript files. You need to install `typescript >= 4.0.0` for this. Type definitions are placed next to their implementation, hand-written `d.ts` files are copied over as is. You can [disable generation](configuration), but we strongly recommend against it — people using your library might use TypeScript, for which they require these type definition files.
-- A `package.json` copied from the project root with all fields except `"scripts"`, `"publishConfig.directory"` and `"publishConfig.linkDirectory"`. The `"dependencies"` field is included, which means you should add packages that you only need for your documentation or demo site to `"devDependencies"`. A `"type": "module"` and an `"exports"` field will be added if it's not defined in the original file.
+To exclude unnecessary files (such as unit tests, or modules that are only imported from `src/routes` etc) you can add them to an `.npmignore` file. This will result in smaller packages that are faster to install.
 
-The `"exports"` field contains the package's entry points. By default, all files in `src/lib` will be treated as an entry point unless they start with (or live in a directory that starts with) an underscore, but you can [configure](configuration) this behaviour. If you have a `src/lib/index.js` or `src/lib/index.svelte` file, it will be treated as the package root.
+Read more about it [here](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#files).
 
-For example, if you had a `src/lib/Foo.svelte` component and a `src/lib/index.js` module that re-exported it, a consumer of your library could do either of the following:
+### exports
+
+The `"exports"` field contains the package's entry points. If you set up a new library project through `npm create svelte@latest`, it's set to a single export, the package root:
+
+```json
+{
+	"exports": {
+		".": {
+			"types": "./dist/index.d.ts",
+			"svelte": "./dist/index.js",
+			"default": "./dist/index.js"
```

Comment:
what is the best way to have "svelte" and "default" condition here? if we use "default", it still requires a "svelte aware" bundler to be usable, so it would break in runtime environments like node, deno if used without a bundler.

This is exactly the same issue we complain about with misconfigured commonjs libraries so i'm not sure we should go there.

Reply:
We decided we want to keep it for a few reasons:
- This is the starter template. People could use it to create a Svelte actions package that has no Svelte components at all. In this case it would be the other way around: they should probably omit the svelte condition but keep the default condition. Since we can't know what the user does, we play it safe and provide the catch-all
- People might have not configured their plugins/bundler correctly yet, because up until know it also worked without for example in the case of webpack telling it that it should recognize the svelte condition
- Whoever knows about all the fine details of this can adjust the exports map accordingly themselves

Reply:
thats the problem though, people don't know how that works and providing default exports that only work when bundled isn't "nice" and contributes to the mess we are in today.

Reply:
For the vast majority of libraries, it's better to distribute the source to avoid Svelte compiler version mismatches. The compiled version only needs to be distributed if you want to use it in a non-Svelte project, which is not the norm. I think what's here now makes sense given that.

Reply:
Just a single anecdote from the field -- but I've struggled a bit to configure a framework-agnostic svelte-package build for internal libraries more than once (@ghostdevv rescued me a few times) -- often when pushing for a team to gradually adopt Svelte.  I think a way to opt-in to a happy path that minimizes vendor-lock could be great for adoption.

Reply:
i fail to understand why it makes sense to export the same file from both svelte condition and default.  
Exporting an index file from "default" that reexports .svelte breaks down without a bundler+svelte. you don't put a svelte file in "main" either. why should exports.default be different?

if you need bundler+svelte and there are no plain js exports, simply don't set the default condition. 
Arguably the default node error message could be nicer if you tried to import it unbundled, but that's not on us. you could export an error.js that throws a nicer message if you wanted to

Reply:
I think Ben is right about most packages only targeting Svelte codebases. However I think it does make sense to only populate the `svelte` field with svelte code and `default` with pure js **if** our tooling will not choke (vite, rollup etc). It's a way I have published packages in the past and for the most part it's worked fine, even if the Svelte tooling has just ended up loading the pre-compiled version (though I do know there are issues with that).

Reply:
My concern with omitting `default` is that apps might be using tooling that understands `pkg.exports` and understands `.svelte` files but not the `svelte` export condition. E.g. if you're using `rollup-plugin-svelte` and you don't set `exportConditions: ['svelte']` in `@rollup/plugin-node-resolve`, things will work just fine if `default` is set, but if it's omitted it will fail. I wouldn't be surprised if there are lots of apps that would break if component libraries omit `default`.

We're not _preventing_ anyone from using `default` to point to compiled code instead of uncompiled code, and anyone who tries to use a Svelte component library that doesn't expose compiled code (most of them) would fail if they weren't using a Svelte plugin anyway, so by including `default` we're just maximising the likelihood that things will work for the largest number of people

Reply:
Anyway this only affects `create-svelte` so we can always revisit the decision later, if we confirm that omitting `default` won't cause breakage in the wild. I think we should merge this PR so we can get people kicking the tyres of the new `svelte-package` ASAP

Reply:
note that i am not advocating for distributing compiled svelte components on the default export either. 
I agree that distributing compiled components is a special case that needs more thinking / careful setup, as it basically ties the published package to the svelte compiler version it was built with in regards to interoperability with other svelte components and it might still cause problems for deduping svelte in some situations if the runtime bits were bundled too.

But as @ghostdevv  pointed out earlier there are libraries out there that offer both preprocessed svelte components and vanilla js code along with them. The vanilla code can be used without a bundler that knows svelte. That is where default or import condition can be useful.

Reply:
going to mark this conversation as resolved, per https://github.com/sveltejs/kit/pull/8922#issuecomment-1432330770